src/bsoncxx/array/value.cpp

```diff
@@ -23,6 +23,13 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace array {
 
+namespace {
+const auto nop_deleter = [](uint8_t*) {};
+}  // namespace
+
+value::value() noexcept : _data(nullptr, nop_deleter), _length() {
```

Comment:
Very interesting question, and I don't know, so I'm going to switch it. Too bad we don't have constexpr lambdas yet.